### DISCUSSION TITLE
Running unit test / generating any possible type of executable

### DISCUSSION THREAD

**Discussion by jrpally** [pyinstaller/pyinstaller#8391] [Help]
Hi, is there any way to run unit tests or do you have a module to generate all kind of pyinstaller exe files possible? 

We run an analyzer for executable files in our company I would like to know if pyinstaller tests can generate any kind of file.

**Comment by bwoodsend** [comment]
Unit tests to test what? What do you mean by *all kind of pyinstaller exe files*?

**Reply by jrpally** [reply]
Hi @bwoodsend I’m working on developing scanners for executables generated by PyInstaller and am keen to understand the range of .exe files it can produce, especially through its test suite. Could you share insights or point me to specific tests that generate various types of executables? I’m interested in differences arising from:

	•	Console vs. GUI application configurations.
	•	Use of specific Python libraries or features.
	•	Platform (Windows, Linux, macOS) or architecture (32 vs. 64 bits) variations.

Any guidance on accessing these executable samples or replicating such tests would be greatly appreciated.

Thanks for your hard work on PyInstaller!

**Reply by rokm** [reply]
If you are interested in difference between *executable* part, it should suffice to look at different variants of bootloader executable. Either the pre-compiled ones at https://github.com/pyinstaller/pyinstaller/tree/develop/PyInstaller/bootloader or the ones you compile yourself.

Anything else is an open-ended problem, and you'll have to build sample programs yourself. 

You can run the test suite using `pytest`. Each test typically cleans up after itself on success, but you can disable that by setting `PYI_BUILDER_CLEANUP` environment variable to 0.
